### ISSUE TITLE
Profiler does not collect any samples after JavaCall.init()

### ISSUE THREAD

**Issue by albertas-jn** [JuliaLang/julia#25371]
After calling JavaCall.init() in JavaCall.jl package, the profiler does not collect any samples anymore until REPL is restarted. Profile.init() returns the same (1000000, 0.001) both before and after the call.

```julia
Profile.clear()
@profile sin.(rand(100_000_000))
Profile.print() 
[Prints collected samples]

Profile.clear()
using JavaCall
JavaCall.init()
@profile sin.(rand(100_000_000))
Profile.print() 
WARNING: There were no samples collected. Run your program longer (perhaps by
running it multiple times), or adjust the delay between samples with Profile.init().
```
--
Julia 0.6.2, JavaCall 0.5.2, Linux (x86_64-pc-linux-gnu)

**Comment by StefanKarpinski** [comment]
cc @aviks – seems likely to be a JavaCall issue, but I'm not sure.

**Comment by albertas-jn** [comment]
If so, does that mean that packages can silently break core functionality?

**Comment by ihnorton** [comment]
Based on the [JavaCall FAQ](https://github.com/JuliaInterop/JavaCall.jl/blob/master/doc/faq.md), I'm pretty sure this is because JVM installs its own signal handlers which interfere with the signal handlers Julia uses for profiling.

> If so, does that mean that packages can silently break core functionality?

Both Julia and Java have FFI. Strictly speaking, there's not much a package can't do within the limits of the kernel. (which on x86 these days is ... well, nevermind).

x-ref https://github.com/JuliaInterop/JavaCall.jl/issues/71

**Comment by vtjnash** [comment]
Yeah, I think is pretty much just a limitation of the linux kernel. There's a way to configure the JVM (I forget if its a flag, library, or environment variable) to use signals differently, which may help.

**Comment by albertas-jn** [comment]
> Strictly speaking, there's not much a package can't do within the limits of the kernel.

So there's no guarantee that some package, or an update of a package, will not cause unrelated code to produce rubbish?

**Comment by KristofferC** [comment]
> So there's no guarantee that some package, or an update of a package, will not cause unrelated code to produce rubbish?

It depends on what you mean with unrelated, but in general no. A package could e.g. redefine plus to mean minus.

**Comment by albertas-jn** [comment]
> A package could e.g. redefine plus to mean minus.

Right, but in this case it will not be silent. I'm wondering about cases without any warning.

**Comment by StefanKarpinski** [comment]
In general, any code you use might cause your program to do anything. This is not unique to Julia.
